src/firefly/js/charts/ui/CombineChart.jsx

```diff
@@ -113,62 +174,95 @@ function createTableModel(showAll, tbl_id) {
     return table;
 }
 
-const CombineChartDialog = ({onComplete}) => {
-
+const ChartSelectionTable = ({tbl_id, chartIds, selectedChartId}) => {
     const [showAll, setShowAll] = useState(false);
+    const showAllHint = 'Show all charts even ones that may not combine well';
+
+    useEffect(() => {
+        const selTbl = createTableModel(chartIds, selectedChartId ?? chartIds[0], showAll, tbl_id);
+        dispatchTableAddLocal(selTbl, undefined, false);
+    }, [showAll]);
+
+    return (
+        <Stack height={125}>
+            {/*--- temporarily removed to only combine charts with similar axes
+                <div style={{display: 'inline-flex', alignItems: 'center'}}>
+                    <label htmlFor='showAll'>Show All Charts: </label>
+                    <input type='checkbox' id='showAll' title={showAllHint} value={showAll} onClick={(e) => setShowAll(e.target?.checked)}/>
+                    <span className='CombineChart__hints'>({showAllHint})</span>
+                </div>
+                ----------*/}
+            <TablePanel {...{tbl_id, showToolbar: false, showUnits: false, showTypes: false}}/>
+        </Stack>
+    );
+};
+
 
+const CombineChartDialog = ({onComplete, chartIds, selectedChartId, allowChartSelection, deriveTraceTitle}) => {
     const tbl_id = 'combinechart-tbl-id';
     const groupKey = 'combinechart-props';
 
+    const [resolvedChartIds, setResolvedChartIds] = React.useState([]);
+    const [rejectedChartReasons, setRejectedChartReasons] = React.useState([]);
     const doCascading = useStoreConnector(()=>getFieldVal(groupKey, 'doCascading'));
+    const tblSelectedChartIds = useStoreConnector(()=>
+        getColumnValues(getSelectedDataSync(tbl_id), 'ChartId'));
 
     useEffect(() => {
-        const selTbl = createTableModel(showAll, tbl_id);
-        dispatchTableAddLocal(selTbl, undefined, false);
-    }, [showAll]);
+        if (chartIds[0] instanceof Promise) {
```

Comment:
May need to handle cases when `chartIds` is undefined to avoid unexpected error.

Reply:
I think I can just filter them out of resolvedChartIds, not sure what to show for error because I won't have error message that it's undefined for which row.

Or maybe I should handle handle the undefined check in fetch function to explicitly throw a Error with all the info/

Reply:
Ah, I see that CombineChartDialog is not exported.  That's how you can expect `chartIds` to have valid values.  Ignore my comment.